pkg/gadgets/sigsnoop/gadget.go

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2019-2022 The Inspektor Gadget authors
+
+package sigsnoop
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/sigsnoop/tracer"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/sigsnoop/types"
+
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/apis/gadget/v1alpha1"
+)
+
+type Trace struct {
+	resolver gadgets.Resolver
+
+	started bool
+	tracer  *tracer.Tracer
+}
+
+type TraceFactory struct {
+	gadgets.BaseFactory
+}
+
+func NewFactory() gadgets.TraceFactory {
+	return &TraceFactory{
+		BaseFactory: gadgets.BaseFactory{DeleteTrace: deleteTrace},
+	}
+}
+
+func (f *TraceFactory) Description() string {
+	return `sigsnoop traces all signals sent on the system.
+
+The following parameters are supported:
+- failed: Trace only failed signal sending (default to false).
+- signal: Which particular signal to trace (default to all).
+- pid: Which particular pid to trace (default to all).
+`
+}
+
+func (f *TraceFactory) OutputModesSupported() map[string]struct{} {
+	return map[string]struct{}{
+		"Stream": {},
+	}
+}
+
+func deleteTrace(name string, t interface{}) {
+	trace := t.(*Trace)
+	if trace.tracer != nil {
+		trace.tracer.Stop()
+	}
+}
+
+func (f *TraceFactory) Operations() map[string]gadgets.TraceOperation {
+	n := func() interface{} {
+		return &Trace{
+			resolver: f.Resolver,
+		}
+	}
+
+	return map[string]gadgets.TraceOperation{
+		"start": {
+			Doc: "Start sigsnoop gadget",
+			Operation: func(name string, trace *gadgetv1alpha1.Trace) {
+				f.LookupOrCreate(name, n).(*Trace).Start(trace)
+			},
+		},
+		"stop": {
+			Doc: "Stop sigsnoop gadget",
+			Operation: func(name string, trace *gadgetv1alpha1.Trace) {
+				f.LookupOrCreate(name, n).(*Trace).Stop(trace)
+			},
+		},
+	}
+}
+
+func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
+	if t.started {
+		gadgets.CleanupTraceStatus(trace)
+
```

Comment:
In case this PR is merged after https://github.com/kinvolk/inspektor-gadget/pull/528, please update it with the same approach.